base/compiler/types.jl

```diff
@@ -76,41 +82,46 @@ function Effects(
     effect_free::TriState,
     nothrow::TriState,
     terminates::TriState,
-    nonoverlayed::Bool)
+    nonoverlayed::Bool,
+    notls::TriState)
     return Effects(
         consistent,
         effect_free,
         nothrow,
         terminates,
         nonoverlayed,
+        notls,
         false)
 end
 
-const EFFECTS_TOTAL    = Effects(ALWAYS_TRUE,      ALWAYS_TRUE,      ALWAYS_TRUE,      ALWAYS_TRUE,      true)
-const EFFECTS_THROWS   = Effects(ALWAYS_TRUE,      ALWAYS_TRUE,      TRISTATE_UNKNOWN, ALWAYS_TRUE,      true)
-const EFFECTS_UNKNOWN  = Effects(TRISTATE_UNKNOWN, TRISTATE_UNKNOWN, TRISTATE_UNKNOWN, TRISTATE_UNKNOWN, true)  # mostly unknown, but it's not overlayed at least (e.g. it's not a call)
-const EFFECTS_UNKNOWN′ = Effects(TRISTATE_UNKNOWN, TRISTATE_UNKNOWN, TRISTATE_UNKNOWN, TRISTATE_UNKNOWN, false) # unknown, really
+const EFFECTS_TOTAL    = Effects(ALWAYS_TRUE,      ALWAYS_TRUE,      ALWAYS_TRUE,      ALWAYS_TRUE,      true, ALWAYS_TRUE)
+const EFFECTS_THROWS   = Effects(ALWAYS_TRUE,      ALWAYS_TRUE,      TRISTATE_UNKNOWN, ALWAYS_TRUE,      true, ALWAYS_TRUE)
```

Comment:
```suggestion
const EFFECTS_THROWS   = Effects(ALWAYS_TRUE,      ALWAYS_TRUE,      TRISTATE_UNKNOWN, ALWAYS_TRUE,      true, false)
```
throwing an exception modifies several task-local states (as well as, obviously, messing with control flow)

Reply:
I agree, but I think I'd like to conceptually separate these out. The way that throwing interacts with task-local state is a bit of an implementation detail and you could imagine a situation where you just immediately wrap the code in a try/catch anyway, in which case you'd probably be ok assuming it doesn't access any other tls state, so I think it's more useful to track this separately.